### PR TITLE
Use decoration hint text as the default value for dropdown button hints

### LINKED ISSUE #111958: DropdownButtonFormField's hintText is not vertically centered if items is null or empty
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Observe that the hintText is off-centered (pushed down vertically).

**Expected results:** hintText should be vertically centered

**Actual results:** hintText is not vertically centered

<details>
<summary>Code sample</summary>

```dart
// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text("Dropdown test"),
        ),
        body: Center(
          child: DropdownButtonFormField<String>(
            decoration: InputDecoration(
              border: OutlineInputBorder(),
              hintText: 'This hint text should be vertically centred',
            ),
            items: [],
            onChanged: (val) {},
          ),
        ),
      ),
    );
  }
}
```

</details>

## Remarks
This seems to only be true for when items == null/empty

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR makes `DropdownButtonFormField` hint defaults to the provided inputDecoration hintText.

Because `DropDownButtonFormField` accepts both a `hint` parameter and a `decoration`parameter, one can expect `InputDecoration.hintText` to be valid.

Before this PR, when `InputDecoration.hintText` was specified, it is shown but the vertical position is wrong.
After this PR, when `InputDecoration.hintText` is specified, it is used as the default value for `DropDownButtonFormField.hint` and `DropDownButtonFormField.disabledHint`.

| Before | After |
|--------|--------|
| image | image |

## Related Issue

Fixes https://github.com/flutter/flutter/issues/111958.

## Tests

Adds 5 tests.

### CODE REVIEW

**Comment by bleroux** [comment]
@justinmc 
To make it easier to review, this PR has 3 commits, the main change is in the first commit. The second commit updates the formatting which makes the diff harder to review.

File: packages/flutter/test/material/dropdown_test.dart [hunk 1]
```diff
@@ -3741,101 +3745,101 @@ void main() {
       alignment: AlignmentDirectional.centerStart,
       isExpanded: false,
     ));
-    expect(tester.getTopLeft(find.text(hintText,skipOffstage: false)).dx, 348.0);
-    expect(tester.getTopLeft(find.text(hintText,skipOffstage: false)).dy, 292.0);
+    expect(tester.getTopLeft(find.text(hintText, skipOffstage: false)).dx, 348.0);
+    expect(tester.getTopLeft(find.text(hintText, skipOffstage: false)).dy, 292.0);
```
Comment (justinmc) [inline packages/flutter/test/material/dropdown_test.dart:3749]: Good catch on these spaces.

**Review by justinmc — APPROVED** [review]
Thank you for the guidance on reviewing by commit, that was much easier! LGTM :+1:

**Review by ebjorklund01 — COMMENTED** [review]
🙌

**Pull request merged**